packages/core/src/Configs/MintV3PoolConfig.class.ts

```diff
@@ -0,0 +1,87 @@
+import { AssetAmount, IAssetAmountMetadata } from "@sundaeswap/asset";
+import { IMintV3PoolConfigArgs } from "../@types/index.js";
+
+import { TFee } from "../@types/queryprovider.js";
+import { Config } from "../Abstracts/Config.abstract.class.js";
+
+export class MintV3PoolConfig extends Config<IMintV3PoolConfigArgs> {
+  assetA?: AssetAmount<IAssetAmountMetadata>;
+  assetB?: AssetAmount<IAssetAmountMetadata>;
+  fees?: TFee;
+  marketTimings?: [bigint, bigint];
+  ownerAddress?: string;
+  protocolFee?: bigint;
+
+  constructor(args?: IMintV3PoolConfigArgs) {
+    super();
+    args && this.setFromObject(args);
+  }
+
+  setFromObject({
+    assetA,
+    assetB,
+    fees,
+    marketTimings,
+    ownerAddress,
+    protocolFee,
+    referralFee,
+  }: IMintV3PoolConfigArgs): void {
+    referralFee && this.setReferralFee(referralFee);
+    this.setAssetA(assetA);
+    this.setAssetB(assetB);
+    this.setFees(fees);
+    this.setMarketTimings(marketTimings);
+    this.setOwnerAddress(ownerAddress);
+    this.setProtocolFee(protocolFee ?? 2_000_000n);
+  }
+
+  buildArgs(): Omit<IMintV3PoolConfigArgs, "protocolFee" | "marketTimings"> & {
+    protocolFee: bigint;
+    marketTimings: [bigint, bigint];
+  } {
+    this.validate();
+    return {
+      assetA: this.assetA as AssetAmount<IAssetAmountMetadata>,
+      assetB: this.assetB as AssetAmount<IAssetAmountMetadata>,
+      fees: this.fees as TFee,
+      marketTimings: this.marketTimings as [bigint, bigint],
+      ownerAddress: this.ownerAddress as string,
+      protocolFee: this.protocolFee ?? 2_000_000n,
+      referralFee: this.referralFee,
+    };
```

Comment:
shouldn't the object values be properly typed already?

Reply:
The configs need to be refactored at some point, but they aren't yet so this is enforced through the validation method.